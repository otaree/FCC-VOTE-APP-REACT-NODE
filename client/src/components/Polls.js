import React from 'react';
import { connect } from 'react-redux';

import PollsList from './PollsList';
import * as pollActions from '../store/actions/poll';
// import "./Polls.css";

export class Polls extends React.Component {
    async componentDidMount() {
        await this.props.fetchPolls();
    }
    render() {
        let polls = (
            <div className="section">
                {
                    this.props.polls.length > 0 ? 
                    (<PollsList polls={this.props.polls} />) :
                    (<div style={{ textAlign: "center" }}>No Polls</div>)
                }
            </div>
        );
        if (this.props.loading) {
            polls = <h2 style={{ textAlign: "center" }}>Loading...</h2>  
        }

        return polls;
    }
}

const mapStateToProps = state => {
    return {
        polls: state.poll.polls,
        loading: state.poll.loading
    };
};

const mapDispatchToProps = dispatch => {
    return {
        fetchPolls: () => dispatch(pollActions.fetchPolls())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Polls);

