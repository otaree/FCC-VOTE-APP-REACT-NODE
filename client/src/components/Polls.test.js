import React from 'react';
import { shallow } from 'enzyme';

import { Polls } from './Polls';


describe("Poll", () => {
    const mockFetchPolls = jest.fn();
    const polls = [];
    const props = {
        fetchPolls: mockFetchPolls,
        polls
    };
    const poll = shallow(<Polls {...props} />);

    it('renders correctly', () => {
        expect(poll).toMatchSnapshot();
    });
});