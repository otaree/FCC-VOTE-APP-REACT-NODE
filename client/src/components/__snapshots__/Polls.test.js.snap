// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Polls
    fetchPolls={[Function]}
    polls={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
                Object {
                        "textAlign": "center",
                      }
        }
>
        No Polls
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No Polls",
        "style": Object {
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": "No Polls",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
                    Object {
                              "textAlign": "center",
                            }
          }
>
          No Polls
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No Polls",
          "style": Object {
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "No Polls",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
